src/electionguard/rsa.c

```diff
@@ -37,7 +37,8 @@ void generate_keys(rsa_private_key* priv_key, rsa_public_key* pub_key)
     /* Select p */
     for(int i = 0; i < BUFFER_SIZE; i++)
     {
-        buffer[i] = RandomSource_get_byte(source) % 0xFF;
+        enum RandomSource_status source_status = RandomSource_get_byte(source, buffer[i]) % 0xFF;
+        assert(source_status == RANDOM_SOURCE_SUCCESS);
```

Comment:
`buffer` is a stack-based array in this function that is uninitialized. This call passes in the value at the `i` index of the `buffer` which isn't what we want since we want a random byte to be written to that location in memory. 

Separately, this new call mods off the enum returned which I don't believe is intended.

Reply:
I think going back to the previous function definition and having the random byte returned makes more sense here. The `% 0xFF` should probably be removed regardless since `0xFF % 0xFF` is 0 so we are losing that as a possible random byte value, and now twice as likely to have 0 as opposed to any other byte value.